plugins/controlboard/src/ControlBoardDriverCurrent.cpp

```diff
@@ -19,29 +19,41 @@ namespace yarp {
 
         bool GazeboYarpControlBoardDriver::getCurrentRange(int j, double *min, double *max)
         {
-            return NOT_YET_IMPLEMENTED("getCurrentRange");
+            std::cout << "max torque " << m_maxTorques[j] << std::endl;
+            if (min && max && j >= 0 && static_cast<size_t>(j) < m_numberOfJoints) {
+                *min = -m_maxTorques[j] / m_kPWM[j];
+                *max = m_maxTorques[j] / m_kPWM[j];
+                return true;
+            }
+            return false;
         }
 
         bool GazeboYarpControlBoardDriver::getCurrentRanges(double *min, double *max)
         {
-            return NOT_YET_IMPLEMENTED("getCurrentRanges");
+            if (!min || !max)) return false;
+            for (size_t j = 0; j < m_numberOfJoints; ++j) {
+                min[j] = -m_maxTorques[j] / m_kPWM[j];
+                max[j] = m_maxTorques[j] / m_kPWM[j];
+            }
+            return true;
         }
 
         bool GazeboYarpControlBoardDriver::setRefCurrent(int j, double v)
         {
-            if (j >= 0 && static_cast<size_t>(j) < m_numberOfJoints) {
-                m_jntReferenceTorques[j] = v*m_kPWM[j];
+            if (!checkIfTorqueIsValid(v * m_kPWM[j]))
+                return false;
+            if (v && j >= 0 && static_cast<size_t>(j) < m_numberOfJoints) {
+                m_jntReferenceTorques[j] = v * m_kPWM[j];
                 return true;
             }
             return false;
         }
 
         bool GazeboYarpControlBoardDriver::setRefCurrents(const int n_joint, const int *joints, const double *t)
-        {
+        {          
```

Comment:
Remember to remove any [trailing whitespace](https://stackoverflow.com/questions/300489/why-is-it-bad-to-commit-lines-with-trailing-whitespace-into-source-control) in your modifications. You can configure KDevelop to automatically remove trailing whitespace, see https://kate-editor.org/2012/10/27/remove-trailing-spaces/ .